src/jq_port.erl

```diff
@@ -202,22 +206,38 @@ kill_port(Port) ->
         {Port, {data, <<"exiting">>}} -> 
             ok
     after TimeToWaitForExitingConfirmation ->
+              PortInfoRes = erlang:port_info(Port, os_pid),
+              case PortInfoRes of
+                  {os_pid, OsPid} when is_integer(OsPid) ->
+                      KillCmdFormat =
+                      case os:type() of
+                          {win32, _} ->
+                              "taskkill /PID ~p /F";
+                          {_, _ } -> 
+                              %% We assume it is UNIX or some other OS
+                              %% with an UNIX kill command.
+                              "kill -9 ~p"
+                      end,
+                      %% This is theoretically unsafe as the process
+                      %% might have died and the its PID got reused.
+                      %% However, on mainstream systems it should not
+                      %% be a problem as it is unlikely that the same
+                      %% PID will get reused very quickly.
+                      os:cmd(io_lib:format(KillCmdFormat, [OsPid]));
+                  _ ->
+                      %% Process should be closed already if we did
+                      %% not get an integer
+                      ok
+              end,
               ok
     end,
-    %% Send close to the port just in case the port does not respond to the exit command for some reason
-    Port ! {self(), close},
     ok.
 
 send_msg_to_port(Port, Msg) ->
     erlang:port_command(Port, Msg).
 
 receive_msg_from_port(Port) ->
-    receive
-        {Port, {data, Data}} ->
-            Data;
-        {'EXIT', Port, Reason} ->
-            {error, port_exited_during_op, Port, Reason}
-    end.
+    receive_msg_from_port(Port, infinity).
```

Comment:
I just wish we won't use `infinity` here, but you decide. 😺

Reply:
will merge now, can resolve this later.